### ISSUE TITLE
training loss does not converge

### ISSUE THREAD

**Issue by sufengniu** [snowflakedb/ArcticTraining#246]
Hello, arctic developer, I used arctic for long context LLM fine tuning (sequence parallelism) training, but I found using arctic training, even overfitting a single sample, the model loss does not decrease (it is jagged), following is my yaml file example:

```
type: sft
micro_batch_size: 8
gradient_accumulation_steps: 1
epochs: 2
activation_checkpoint_cpu_offload: true
tiled_mlp_compute: true

sequence_parallel_size: 8
deepspeed:
  zero_optimization:
    stage: 3
    offload_optimizer:
      device: cpu
    reduce_bucket_size: 1e6
  seq_parallel_communication_data_type: bf16

optimizer:
  type: cpu_adam
  learning_rate: 0.00001

model:
  type: "liger"
  name_or_path: Qwen/Qwen3-32B

  attn_implementation: flash_attention_2
  peft_config:
    peft_type: "Lora"
    target_modules: ["q_proj", "k_proj", "v_proj", "o_proj", "gate_proj", "up_proj", "down_proj"]
    r: 16
    lora_alpha: 32
    bias: "none"
    task_type: "CAUSAL_LM"

data:
  sources:
    - type: huggingface_instruct
      name_or_path: my_local_data_path
      sample_count: 1000
      split: train_sft
      role_mapping:
        system: system
        user: user
        assistant: assistant

  cache_dir: data-cache

  dl_num_workers: 1

  max_length: 250_000
  mask_inputs: true

logger:
  level: WARNING
#  level: INFO

  output_dir: "logs"
  #file_output_ranks: [0,1]
  print_output_ranks: [0,1,2,3,4,5,6,7]
```
the data is a single sample, and I replicate it 1000 times, for each sample, it contains columns user, system, assistant, where system is a very long context (about 125k token size). the loss I observe during training is something like:
Loss: 0.86718
Loss: 0.86718
Loss: 0.87254
...
Loss: 0.86079

Anyone ever experienced such issues?
